### PR TITLE
feat(fastly_alert): implement resource and documentation

### PR DESCRIPTION
**Pull request by darin-nee**
This PR implements a new `fastly_alert` resource.

### CODE REVIEW

**Comment by Integralist** [comment]
👋🏻 You'll need to run `make generate-docs` locally and then commit the changes.

File: fastly/resource_fastly_alert.go [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+		DeleteContext: resourceFastlyAlertDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Text body included in Alert notifications.",
```
Comment (Integralist) [inline fastly/resource_fastly_alert.go]: This kinda read weird. Can we make these descriptions align (where possible) with the API docs?

For example...

```suggestion
				Description: "Additional text that is included in the alert notification.",
```

File: docs/resources/alert.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+### Optional
+
+- `description` (String) Text body included in Alert notifications.
+- `dimensions` (Block List, Max: 1) Dimension filters applied to the metric. (see [below for nested schema](#nestedblock--dimensions))
+- `integration_ids` (Set of String) Set of IDs of integrations that Alert notifications will be sent to.
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
```
Comment (jsonroy-fastly) [inline docs/resources/alert.md:62]: @darin-nee There is also `object` = "definition" that is returned in the response, also read-only

**Review by jsonroy-fastly — COMMENTED** [review]
Great work on this and the go-fastly changes!
Small comment added

File: docs/resources/alert.md (hunk 2 above)
Comment (darin-nee) [reply docs/resources/alert.md:62]: Yep, that's in the API response and elided in the Terraform schema.

**Review by Integralist — COMMENTED** [review]
Sorry, looked like I had a comment from 5 days ago sitting as "pending" 😬

**Comment by Integralist** [comment]
Would also be good to get this added to https://github.com/fastly/terraform-provider-fastly/blob/main/tests/interface/main.tf

**EDIT:** See https://github.com/fastly/terraform-provider-fastly/blob/main/tests/interface/README.md

File: fastly/resource_fastly_alert.go (hunk 1 above)
Comment (darin-nee) [reply fastly/resource_fastly_alert.go]: Yes, done.

**Review by Integralist — APPROVED** [review]
It is always a pleasure reviewing your PRs 👍🏻 You write good, clean code and 99% of the time you just get it "right" the first time. Well done and thanks for making it easy to review 🙂

**Comment by darin-nee** [comment]
Thanks @Integralist , and thanks for the review!

**Review by Integralist — APPROVED** [review]

**Pull request merged**